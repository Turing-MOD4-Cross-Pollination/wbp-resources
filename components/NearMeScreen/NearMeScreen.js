import React, { useState, useEffect } from 'react';
import { StyleSheet, Text, View, Platform } from 'react-native';
import MapView from 'react-native-maps';
import * as Location from 'expo-location';
import * as Permissions from 'expo-permissions';
import { connect } from 'react-redux';

const NearMeScreen = ({ allMeetings }) => {
  const [location, updateLocation] = useState(null);
  const [errorMessage, updateError] = useState(null);
  const [loaded, updateLoaded] = useState(false);

  const getLocationAsync = async () => {
    const { status } = await Permissions.askAsync(Permissions.LOCATION);
    if (status !== 'granted') {
      updateError('Permission to access location was denied');
      updateLoaded(true);
    } else {
      const newLocation = await Location.getCurrentPositionAsync({
        enableHighAccuracy: true,
      });
      updateLocation(newLocation);
      updateLoaded(true);
      updateError(null);
    }
  };

  useEffect(() => {
    const checkDevice = () => {
      if (Platform.OS === 'android') {
        updateError(
          'Oops, this will not work on Sketch in an Android emulator. Try it on your device!',
        );
        updateLoaded(true);
      } else {
        getLocationAsync();
      }
    };

    checkDevice();
  }, []);

  if (loaded) {
    if (errorMessage) {
      return (
        <>
          <Text style={styles.header}>Meetups Near Me</Text>
          <View style={styles.container}>
            <Text>{JSON.stringify(errorMessage)}</Text>
          </View>
        </>
      );
    }
    if (location) {
      return (
        <>
          <Text style={styles.header}>Meetups Near Me</Text>
          <MapView
            style={styles.mapStyle}
            showsUserLocation
            region={{
              latitude: location.coords.latitude,
              longitude: location.coords.longitude,
              latitudeDelta: 0.1,
              longitudeDelta: 0.1,
            }}
          >
            {allMeetings &&
              allMeetings.map((meeting, index) => {
                const { latitude, longitude, title, subtitle } = meeting;
                return (
                  <MapView.Marker
                    key={latitude + index}
                    coordinate={{ latitude, longitude }}
                    title={title}
                    description={subtitle}
                  />
                );
              })}
          </MapView>
        </>
      );
    }
  } else {
    return (
      <>
        <Text style={styles.header}>Meetups Near Me</Text>
        <View style={styles.container}>
          <Text>Waiting...</Text>
        </View>
      </>
    );
  }
};

NearMeScreen.navigationOptions = () => ({
  title: 'Near Me',
  headerStyle: {
    backgroundColor: '#4e5b31',
    elevation: 0,
    shadowOpacity: 0,
    borderBottomWidth: 0,
  },
  headerTintColor: '#fff',
  headerTitleStyle: {
    fontWeight: 'bold',
  },
});

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#ecf0f1',
  },
  mapStyle: {
    flex: 1,
  },
  paragraph: {
    margin: 24,
    fontSize: 18,
    textAlign: 'center',
  },
  header: {
    fontSize: 32,
    paddingBottom: 15,
    paddingTop: 20,
    paddingRight: 15,
    paddingLeft: 15,
    backgroundColor: '#4e5b31',
    color: 'white',
    fontWeight: 'bold',
  },
});

export const mapStateToProps = (state) => ({
  allMeetings: state.allMeetings,
});

export default connect(mapStateToProps)(NearMeScreen);
