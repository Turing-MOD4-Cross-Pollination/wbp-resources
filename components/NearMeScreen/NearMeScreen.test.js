// import React from 'react';
// import renderer from 'react-test-renderer';
// import { mount, shallow } from 'enzyme';
// import NearMeScreen from './NearMeScreen';
// import 'babel-polyfill';

// jest.mock('react-native', () => ({
//   Platform: () => ({
//     OS: 'iOS',
//   }),
//   StyleSheet: () => ({
//     create: jest.fn(),
//   }),
// }));

// jest.mock('react-native-maps', () => ({
//   UIManager: () => ({
//     OS: 'iOS',
//   }),
// }));

// describe('NearMeScreen', () => {
//   it('renders correctly', () => {
//     const tree = renderer.create(<NearMeScreen />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });

// describe('NearMeScreen', () => {
//   let Platform;
//   beforeEach(() => {
//     Platform = require('react-native').Platform;
//     Platform.OS = 'ios';
//   });

//   it('renders correctly', () => {
//     const wrapper = shallow(<NearMeScreen />);
//     expect(wrapper).toMatchSnapshot();
//   });
// });

import React from 'react';
import { shallow } from 'enzyme';

import NearMeScreen from './NearMeScreen';

// const Platform = require('react-native');

// Platform.OS = 'ios';

jest.mock('expo-location', () => ({
  UIManager: () => ({
    OS: 'iOS',
  }),
}));

jest.mock('expo-permissions', () => ({
  UIManager: () => ({
    OS: 'iOS',
  }),
}));

describe('NearMeScreen', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NearMeScreen />);
  });

  it('App should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
