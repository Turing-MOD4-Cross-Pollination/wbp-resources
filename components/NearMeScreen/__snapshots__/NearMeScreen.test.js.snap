// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NearMeScreen App should match snapshot 1`] = `ShallowWrapper {}`;
